Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -144,4 +176,67 @@ namespace ROS2
     {
         return m_defaultSpawnPose;
     }
+
+    void ROS2SpawnerComponent::GetSpawnPointsNames(
+        const ROS2::GetSpawnPointsNamesRequest request, ROS2::GetSpawnPointsNamesResponse response)
+    {
+        for (auto spawn_point : GetSpawnPoints())
+        {
+            response->model_names.emplace_back(spawn_point.first.c_str());
+        }
+    }
+
+    void ROS2SpawnerComponent::GetSpawnPointInfo(const ROS2::GetSpawnPointInfoRequest request, ROS2::GetSpawnPointInfoResponse response)
+    {
+        const AZStd::string_view key(request->model_name.c_str(), request->model_name.size());
+
+        auto spawn_points = GetSpawnPoints();
+        if (spawn_points.contains(key))
+        {
+            auto info = spawn_points.at(key);
+
+            geometry_msgs::msg::Pose pose;
+            pose.position.x = info.pose.GetTranslation().GetX();
+            pose.position.y = info.pose.GetTranslation().GetY();
+            pose.position.z = info.pose.GetTranslation().GetZ();
+            pose.orientation.x = info.pose.GetRotation().GetX();
+            pose.orientation.y = info.pose.GetRotation().GetY();
+            pose.orientation.z = info.pose.GetRotation().GetZ();
+            pose.orientation.w = info.pose.GetRotation().GetW();
```

Comment:
Maybe some conversion functions would be in order here. There is a file with this kind of util and it might be added there